### DISCUSSION TITLE
Standing Ovation goes to you sir...

### DISCUSSION THREAD

**Discussion by getcheffy** [madprops/grasshopper#112] [General]
This is ONE HELL OF AN ADD ON. like dayummm. just got it today, and just started playing with it an hour ago and man oh man this thing is ROBUST to say the least. It makes Foxy Tabs look like the JV squad.  The only thing (so far... haha I will prob write again if I think of something) is I was there was a pagenation/pagerizing ability. 

truthfully all you need to do is add the ability to add  user scirpts, automation. block then close tab, but most importantly, Session saving akin to session buddy a la chrome. but like, whatever, this thing is already so amazing I'm amazed at the things you can do with it, that seem so simple and DUH, and like why haven't of the big 3 done something like this yet. OOOH they finally added vertical tabs, how ground breaking. like gtfo. why did it take 20 years for that when extensions had been out for 10. haha anyway just wanted to come and say Bravo, and tip my cap to you good sir. and most of all thanks for making it free, so many dirtbags try and charge for extensions. as an OG ext user self dubbed the Extension Emperor,  I have tried at least 5k extensions at this point, and it really irked me when I started to see a few beginning to charge for their extensions. 

Put your dono link via a coffee icon somewhere on the ext if you haven't already and I missed it. I'm sure people will donate big time once they use this. I will post and share this gem on twitter and places. 

just TLDR, Great Job dude, S+ ranking on this. its now a must have mando install on new installs. ublock, CLUT, PREVUE, and now this.  if you haven't used PREVUE, you are missing out on how the internet should be perused. It speeds up everything drastically and I literally cant use the web without it anymore. 

CHeers!

**Comment by madprops** [comment]
Hey thanks! I'm glad you find Grasshopper useful, I use it myself on all my firefox profiles. My main focus now is to fix bugs when they appear, I don't plan on adding any big features for now. However I note what you say about pagination, automation, and sessions. For pagination (if I guess what you meant) I recently added a similar ability by using PageUp and PageDown keys, it should go down/up pages at specific points.  Automation/user-scripts is something to think about. For sessions, there used to be something like that, but I decided I'd prefer to just let extensions dedicated to that handle that since it's a bit messy. Thanks for the kind words. Right now there's no donation link I might set that up eventually. 

![Red-legged_Grasshopper_(Melanoplus_femurrubrum)_(43478105211)](https://github.com/user-attachments/assets/f38d9b8d-48d0-4328-bd51-d456f7f6cad4)

**Comment by madprops** [comment]
Btw thanks for the Prevue tip, I like how I can just drag links to trigger it, very nice.
